File: src/main/java/io/github/support/terminal/platform/app/domains/core/api/telegram/responses/UpdateResponse.java
package io.github.support.terminal.platform.app.domains.core.api.telegram.responses;

import com.fasterxml.jackson.annotation.JsonProperty;
import io.github.support.terminal.platform.app.domains.core.api.telegram.models.Update;
import lombok.Data;
import lombok.ToString;
import lombok.experimental.Accessors;

import java.util.Collection;

@Data
@ToString
@Accessors(chain = true)
public class UpdateResponse {

    @JsonProperty("ok")
    private Boolean ok;

    @JsonProperty("result")
    private Collection<Update> result;

}

File: src/main/kotlin/io/github/bot/terminal/application/domains/integrations/telegram/responses/UpdateResponse.kt
package io.github.bot.terminal.application.domains.integrations.telegram.responses

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import io.github.bot.terminal.application.domains.integrations.telegram.models.Update

@JsonIgnoreProperties(ignoreUnknown = true)
data class UpdateResponse(
        @JsonProperty("ok") val ok: Boolean,
        @JsonProperty("result") val result: Collection<Update> = listOf()
)